tests/unit/conftest.py

```diff
@@ -134,3 +142,45 @@ def validate_target_transpile(input_sql, *, target=None, pretty=False):
 @pytest.fixture(scope="session")
 def dialect_context():
     yield validate_source_transpile, validate_target_transpile
+
+
+def parse_sql_files(input_dir: Path, source: str, target: str, is_expected_exception):
+    suite = []
+    for dirpath, _, filenames in os.walk(input_dir):
+        if dirpath.endswith("__pycache__"):
+            continue
+        for file in filenames:
+            if not file.endswith('.sql'):
+                continue
+            abs_path = Path(dirpath) / file
+            with open(abs_path, 'r', encoding="utf-8") as file_content:
+                content = file_content.read()
+            if content:
+                parts = content.split(f"-- {source.lower()} sql:")
```

Comment:
this gets very complicated and imposes restrictions on what the format should be. we need to limit format to have just `-- databricks sql:` and split by it. if variant type needs checking, then put all variant related queries in a separate folder and read them in a separate bulk